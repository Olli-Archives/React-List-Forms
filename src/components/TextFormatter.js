import React, { PureComponent } from 'react';
import TextInputForm from './TextInputForm';
import PrintMessage from './PrintMessage';


export default class TextFormatter extends PureComponent {
  state = {
    text: '',
    message: ''
  }

  handleSubmit = (e) => {
    e.preventDefault();
    this.setState({ message: this.state.text });
    this.setState({ text: '' });
  }
  handleTextChange = ({ target }) => {
    const text = target.value;
    this.setState({ text: text });
  }

  render() {
    const { text, message } = this.state;
    return (
      <>
        <TextInputForm text={text} handleTextChange={this.handleTextChange} handleSubmit={this.handleSubmit} />
        <PrintMessage message={message}/>
      </>
    );
  }

}
