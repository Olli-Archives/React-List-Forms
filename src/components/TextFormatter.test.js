import TextFormatter from './TextFormatter';
import { shallow } from 'enzyme';
import React from 'react';

describe('TextFormatterComponent', ()=>{
  it('renders text on submit', ()=>{
    const wrapper = shallow(<TextFormatter/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
