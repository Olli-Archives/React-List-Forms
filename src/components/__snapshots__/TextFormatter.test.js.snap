// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFormatterComponent renders text on submit 1`] = `
<Fragment>
  <TextInputForm
    handleSubmit={[Function]}
    handleTextChange={[Function]}
    text=""
  />
  <PrintMessage
    color="purple"
    message=""
  />
  <ColorOptions
    handleColorChange={[Function]}
  />
</Fragment>
`;
